synapse/storage/e2e_room_keys.py

```diff
@@ -236,6 +246,7 @@ def _get_e2e_room_keys_version_info_txn(txn):
                 ),
             )
             result["auth_data"] = json.loads(result["auth_data"])
+            result["version"] = str(result["version"])
```

Comment:
Can we not just keep this as an int?

Reply:
Is there a reason to then not have them as strings in the database layer?

Reply:
It currently does a MAX() at the db layer to get the largest version, then increments it in python, which breaks when the column type is a string because "10" < "2" which is the bug this fixes.

Reply:
why does the API have it as a string when JSON has numbers?

Reply:
#e2e-dev:matrix.org has the API / spec discussion: this one was: https://matrix.to/#/!uewiilduiDRfPomIha:matrix.org/$15397878202555374QpBig:matrix.org